central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -372,8 +444,36 @@ func (s *serviceImpl) GetComplianceProfileCheckDetails(ctx context.Context, requ
 	if err != nil {
 		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results for query %v", parsedQuery)
 	}
+	if len(scanResults) == 0 {
+		return nil, nil
+	}
+
+	rules, err := s.ruleDS.SearchRules(ctx, search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorRuleRef, scanResults[0].GetRuleRefId()).ProtoQuery())
```

Comment:
[nit] Should we check here that the length of `scanResults` is exactly `1`? Similarly to what you do in line 455 with rules.

Reply:
Since the data is denormalized and these searches are on name, we  can get more than 1 as the name would occur across all clusters.  But since the profile version is in the name of the check result we will only get one rule that should match across all clusters.

I probably could have abstracted that to the datastore with some distinct lookup, but I need to get finished and make things better later.